taskvine/src/worker/vine_cache.c

```diff
@@ -69,9 +70,79 @@ struct vine_cache * vine_cache_create( const char *cache_dir )
 }
 
 /*
-Delete the cache manager structure, though not the underlying files.
+Load existing cache directory into cache structure.
+*/
+void vine_cache_load(struct vine_cache *c)
+{
+	DIR *dir = opendir(c->cache_dir);
+	if(dir){
+		debug(D_VINE, "loading cache at: %s", c->cache_dir);
+		struct dirent *d;
+		while((d=readdir(dir))){
+			debug(D_VINE, "found %s in cache at: %s",d->d_name, c->cache_dir);
+			struct stat info;
+			int64_t nbytes, nfiles;
+			char *cache_path = vine_cache_full_path(c,d->d_name);
+			if(stat(cache_path, &info)==0){
+				if(!strncmp(d->d_name, "file", strlen("file"))){
```

Comment:
Can you make this even simpler?
It seems like the same action is taken no matter what the name pattern is.
And for that matter, the manager might change the naming pattern, and I'm not sure that the worker should care.

Reply:
This has been condensed.